enhancements/storage/cinder-csi-driver-operator-configuration.md

```diff
@@ -0,0 +1,349 @@
+[OCCDO]: https://docs.openshift.com/container-platform/4.9/storage/container_storage_interface/persistent-storage-csi-cinder.html
+
+### Separate ConfigMaps for legacy cloud provider and Cinder CSI driver
+
+The cloud configuration is read by multiple components which will not all be
+upgraded simultaneously. If we simply updated the
+`openshift-config-managed/kube-cloud-config` to be compatible with the Cinder
+CSI driver, we would risk breaking KCM and kubelets which have not yet been
+upgraded. Therefore we are constrained to write two versions of the managed
+config map.
+
+CCO will be continue to create `openshift-config-managed/kube-cloud-config`,
+which will remain compatible with the legacy cloud provider (and by extension,
+the legacy block storage service). OCCDO will create a new
+`openshift-cluster-csi-drivers/cloud-conf` config map, which will be compatible
+with the Cinder CSI driver. The Cinder CSI driver will then read its
+configuration from `openshift-cluster-csi-drivers/cloud-conf`.
+
+In a future release, we can deprecate and remove CCO's management of
+`openshift-config-managed/kube-cloud-config` and eventually the config map
+itself. We can also remove the `openstack-credentials/openstack-credentials`
+secret, which will no longer be used by anything. This effort will be
+accomplished as part of the migration to CCM.
+
+### Preventing upgrade when config cannot be upgraded
+
+If OCCDO is not able to generate `openshift-cluster-csi-drivers/cloud-conf` from
+`openshift-config/cloud-provider-config`, it will mark itself as non-upgradable
+with a user-actionable error message explaining how
+`openshift-config/cloud-provider-config` is incompatible.
+
+### User Stories
+
+#### Story 1 - Fresh install
+
+The Tyrell Corporation wants to deploy OCP 4.11 on their OSP cloud. OCP
+automatically configures the Cinder CSI driver out-of-the-box with
+configuration native to this CSI driver.
+
+#### Story 2 - Upgrade with default configuration
+
+The Wallace Corporation wants to upgrade their OCP 4.10 cluster to OCP 4.11.
+They are using the default cloud provider configuration. Upon upgrading, their
+cluster should automatically configure the Cinder CSI driver following the
+upgrade, and the upgrade should require no additional user input.
+
+#### Story 3 - Upgrade with non-default compatible configuration
+
+The Acme Corporation wants to upgrade their OCP 4.10 cluster to OCP 4.11.
+The have configure a `openshift-config/cloud-provider-config` that contains:
+
+```ini
+[Global]
+secret-name = openstack-credentials
+secret-namespace = kube-system
+
+[BlockStorage]
+trust-device-path = true
+```
+
+After upgrading their cluster will automatically configure the Cinder CSI
+driver with no additional user input required. After the upgrade, their cluster
+should continue to function with legacy configuration options removed.
+Specifically, Cinder CSI driver will be using the following configuration:
+
+```ini
+[Global]
```

Comment:
so the management of these two sections could be removed from CCCMO?

Reply:
We need to keep the `[Global]` section in both since it provides information about the `clouds.yaml` file, which in turn provides authentication information. We can remove (and have removed) management for the `[BlockStorage]` section in CCCMO. Is that what you mean?